pilot/cmd/pilot-discovery/mock/service.go

```diff
@@ -32,10 +32,18 @@ var (
 		Protocol:             model.ProtocolHTTP,
 		AuthenticationPolicy: meshconfig.AuthenticationPolicy_INHERIT,
 	}
+	ExtHTTPService = MakeExternalHTTPService("httpbin.default.svc.cluster.local",
+		"httpbin.org", "")
+	ExtHTTPSService = MakeExternalHTTPSService("httpsbin.default.svc.cluster.local",
+		"httpbin.org", "")
 	Discovery = &ServiceDiscovery{
 		services: map[string]*model.Service{
-			HelloService.Hostname: HelloService,
-			WorldService.Hostname: WorldService,
+			HelloService.Hostname:   HelloService,
+			WorldService.Hostname:   WorldService,
+			ExtHTTPService.Hostname: ExtHTTPService,
+			// TODO external https is not currently supported - this service
+			// should NOT be in any of the .golden json files
+			ExtHTTPSService.Hostname: ExtHTTPSService,
```

Comment:
nit. This code should go away. We only do the clunky http://google:443 for our external services. For kube external services, its no different from any other kube service (i.e. http/https should be allowed). So we should remove the implicit conversion from https to http.

Reply:
Will be handled by https://github.com/istio/istio/issues/2844 in a separate PR.